### PR TITLE
create input with mui and storybook

### PR DESCRIPTION
**Pull request by lielsa2023**
A generic component of an input field called MyInput that accepts various parameters
and can be presented and learned through a storybook

### CODE REVIEW

**Review by CohenAvigail — APPROVED** [review]

File: client-side/src/stories/genericInput.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TextField from '@mui/material/TextField';
```
Comment (e171271) [inline client-side/src/stories/genericInput.jsx:3]: please collect all the material imports to on.
like this:
import {TextField, InputAdornment } from '@mui/material';

File: client-side/src/stories/genericInput.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import InputAdornment from '@mui/material/InputAdornment';
+import '../style/genericInput.scss';
+
+const GenericInput = ({ label, type='text', value, onChange, size='medium', width='20%',
```
Comment (e171271) [inline client-side/src/stories/genericInput.jsx:7]: all props that you didn't defined as isRequired should have a default value here

File: client-side/src/stories/genericInput.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  const handleChange = (e) => {
+    if (onChange) {
+      onChange(e.target.value);
+    }
+  };
+
+  let error = false;
+  let helperText = '';
+
+  if (validation && typeof validation === 'function') {
```
Comment (e171271) [inline client-side/src/stories/genericInput.jsx:18]: we can't write code in the root of the component, you should put it inside some function

File: client-side/src/stories/genericInput.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+const GenericInput = ({ label, type='text', value, onChange, size='medium', width='20%',
+   icon: Icon, validation, ...rest }) => {
+  const handleChange = (e) => {
+    if (onChange) {
+      onChange(e.target.value);
+    }
+  };
+
+  let error = false;
```
Comment (e171271) [inline client-side/src/stories/genericInput.jsx:15]: those variables should be state variables

File: client-side/src/stories/genericInput.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  if (validation && typeof validation === 'function') {
+    const validationResult = validation(value);
+    if (validationResult && validationResult.error) {
+      error = true;
+      helperText = validationResult.helperText || 'Invalid input';
+    }
+  }
+
+  const inputStyle = {
+    width: width
```
Comment (e171271) [inline client-side/src/stories/genericInput.jsx:27]: {width: width} is same as {width} please use this

File: client-side/src/stories/genericInput.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        style={inputStyle}
+        {...rest}
+      />
+    </div>
+  );
+};
+
+GenericInput.propTypes = {
+  label: PropTypes.string.isRequired,
+  type: PropTypes.string,
```
Comment (e171271) [inline client-side/src/stories/genericInput.jsx:57]: type should be an enum - you have only specific types that could be send in this props

File: client-side/src/stories/genericInput.scss [hunk 7]
```diff
@@ -0,0 +1,61 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&display=swap');
+
+
+
+.generic-input .MuiOutlinedInput-root {
+  border-radius: 8px;
+  transition: border-color 0.3s ease, box-shadow 0.3s ease;
+  border-color: rgb(61, 192, 155);
```
Comment (e171271) [inline client-side/src/stories/genericInput.scss:8]: all colors should be defined as a variables

File: client-side/src/stories/genericInput.scss [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    font-size: 14px;
+    transition: color 0.3s ease;
+  }
+
+  &.Mui-error {
+    border-color: red;
+  }
+}
+
+.generic-input .MuiFormLabel-root {
```
Comment (e171271) [inline client-side/src/stories/genericInput.scss:39]: please use scss hierarchic - this block could be with the previous one.
you should write something like:
 .generic-input {
   .MuiOutlinedInput-root {
      ...all the prev styles
   }
  ..MuiFormLabel-root{
      ...all the current styles
   }
}

File: client-side/src/stories/genericInput.scss [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  font-weight: 400;
+  font-size: 16px;
+  transition: color 0.3s ease;
+
+  &.Mui-focused {
+    color: rgb(45, 158, 126);
+  }
+}
+
+.generic-input .MuiFormHelperText-root {
```
Comment (e171271) [inline client-side/src/stories/genericInput.scss:51]: same as the prev comment, please use the correct hierarchic

File: client-side/src/stories/genericInput.stories.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
+
+import React from 'react';
+import GenericInput from '../inputs/genericInput';
```
Comment (e171271) [inline client-side/src/stories/genericInput.stories.js:3]: please re-order the imports

**Review by e171271 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**